xbmc/pvr/PVRGUIInfo.cpp

```diff
@@ -284,7 +286,22 @@ void CPVRGUIInfo::UpdateTimeshift(void)
   m_iStartTime = iStartTime;
   m_iTimeshiftStartTime = iStartTime + iMinTime;
   m_iTimeshiftEndTime = iStartTime + iMaxTime;
-  m_iTimeshiftPlayTime = iStartTime + iPlayTime;
+
+  if (m_iTimeshiftEndTime > m_iTimeshiftStartTime)
+  {
+    // getstreamtimes api
+    m_iTimeshiftPlayTime = iStartTime + iPlayTime;
+  }
+  else if (CServiceBroker::GetDataCacheCore().GetSpeed() == 1.0)
+  {
+    // legacy api: playing timeshifted or live
+    m_iTimeshiftPlayTime = std::time(nullptr) - m_iTimeshiftOffset;
+  }
+  else
+  {
+    // legacy api: paused/seeking
+    m_iTimeshiftOffset = std::time(nullptr) - m_iTimeshiftPlayTime;
```

Comment:
this is a code path that is not allowed. if an addon supports timeshift it MUST implement GetStreamTimes. if an addon has not implemented GetStreamTimes, we should disable all timeshift capabilities like pause and seek

Reply:
Okay, that makes it even more simple. Will change.